### ISSUE TITLE
Mininet VM Support on ARM64/Apple Silicon Request

### ISSUE THREAD

**Issue by ez4bk** [mininet/mininet#1155]
### Expected/Desired Behavior:
Mininet VM should be runnable on an M1 Mac machine as well.
### Actual Behavior:
Any VM software would report that only x86 architecture is supported.

**Comment by lantz** [comment]
It would be nice to have an ARM64 VM.

Until then, you may wish to try qemu or utm, both of which should be able to run an x86 VM.

**Comment by paaguti** [comment]
In general, it would be good to have mininet compiled under ARM and ARM64, regardless of the fact that it runs in a VM or on a board/computer/etc. I have been exploring this, and the main point for newer versions in that some components depend on Intel specific libraries.
I didn't have the time to explore how critical these libraries are.
From what I saw when I browsed through them, they mainly dealt with optimisations. Could we live without them, given that this is an emulator. At least I expect functionality and not speed in this environment.

**Comment by Gabriele-tomai00** [comment]
I have a MacBook Pro m1 and tried to compile for arm Linux (using Ubuntu and Fedora for arm), but I have a problem with pyflakes and pyflakes3 packages. I need a version for arm

**Comment by root-hbx** [comment]
I use a MacBook Pro m2, and I face the same issue as well.

I tried to use a VM (in fact, it's orbstack) of Linux 22.04LTS to solve the problem.

However, it's still a problem: ( I follow the steps in a python3 venv)
When I follow the steps on [download](https://mininet.org/download/), I met a problem:

he following packages were automatically installed and are no longer required:
  libperl5.34 libtiff5 perl-modules-5.34
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 287 not upgraded.
Installing Mininet core
/Users/huluobo/mininet /Users/huluobo
cc -Wall -Wextra  \
-DVERSION=\"`PYTHONPATH=. python -B bin/mn --version 2>&1`\" mnexec.c -o mnexec
<command-line>: warning: missing terminating " character
mnexec.c: In function ‘main’:
<command-line>: error: missing terminating " character
mnexec.c:208:28: note: in expansion of macro ‘VERSION’
  208 |             printf("%s\n", VERSION);
      |                            ^~~~~~~
mnexec.c:208:35: error: expected expression before ‘)’ token
  208 |             printf("%s\n", VERSION);
      |                                   ^
make: *** [Makefile:50: mnexec] Error 1

Actually, I do need a version for arm.

**Comment by amirreza225** [comment]
I need it too, Please consider this problem critical!

**Comment by MarlovMrN** [comment]
Agreed! Luckily I have an old desktop sitting here, otherwise I would have to buy an additional pc just to use mininet for my graduation paper, as I can't use my MacOS setup. 

It should be just a matter of using an arm base image and guarantee that all dependent packages can run in arm as well, and if they don't, considering an alternative

**Comment by davidlin2k** [comment]
> Agreed! Luckily I have an old desktop sitting here, otherwise I would have to buy an additional pc just to use mininet for my graduation paper, as I can't use my MacOS setup.
> 
> It should be just a matter of using an arm base image and guarantee that all dependent packages can run in arm as well, and if they don't, considering an alternative

I've tried using the arm based ubuntu image with UTM and it seems to compile for me when I installed from source. Running Mininet in Docker also worked for my Mac. I've shared the VM here maybe give it a try https://github.com/davidlin2k/mininet-docker/releases/tag/2.3.0

**Comment by paaguti** [comment]
If you are on Ubuntu 22.04, you can go directly for the .deb package provided by them. I have a larger group of people working with mininet on a VM in Mac Mx's this way.

Best, /PA

**Comment by amirreza225** [comment]
Hi everyone,

I’ve created a solution for running Mininet on Apple Silicon (M1/M2) Macs using Docker, which addresses the problem discussed here. You can find it at:

👉 https://github.com/amirreza225/MinininetDocker-AppleSiliconCompatible

This project sets up a Mininet environment inside a Docker container that is compatible with Apple Silicon, avoiding the qemu issues and VM hassles. It’s designed to be easy to use and reproducible.

Hope this helps anyone who comes across this thread in the future!

Best,
Amirreza
